src/v/kafka/server/handlers/configs/config_utils.h

```diff
@@ -182,27 +185,160 @@ ss::future<std::vector<R>> do_alter_topics_configuration(
-    // errors
+do_alter_broker_configuartion(request_context& ctx, std::vector<T> resources) {
     std::vector<R> responses;
     responses.reserve(resources.size());
-    std::transform(
-      resources.begin(),
-      resources.end(),
-      std::back_inserter(responses),
-      [](T& resource) {
-          return make_error_alter_config_resource_response<R>(
-            resource,
-            error_code::invalid_config,
-            fmt::format(
-              "changing '{}' broker property isn't currently supported",
-              resource.resource_name));
-      });
 
-    return ss::make_ready_future<std::vector<R>>(std::move(responses));
+    // If central config is disabled, we cannot set broker properties
+    if (!config::node().enable_central_config()) {
+        std::transform(
+          resources.begin(),
+          resources.end(),
+          std::back_inserter(responses),
+          [](T& resource) {
+              return make_error_alter_config_resource_response<R>(
+                resource,
+                error_code::invalid_config,
+                fmt::format(
+                  "changing '{}' broker property isn't currently supported",
+                  resource.resource_name));
+          });
+
+        co_return responses;
+    }
+
+    for (const auto& resource : resources) {
+        cluster::config_update_request req;
+
+        if (!resource.resource_name.empty()) {
+            responses.push_back(make_error_alter_config_resource_response<R>(
+              resource,
+              error_code::invalid_config,
+              "Setting broker properties on named brokers is unsupported"));
+            continue;
+        }
+
+        bool errored = false;
+        for (const auto& c : resource.configs) {
+            auto mapped_name = map_config_name(c.name);
+            if constexpr (std::
+                            is_same_v<T, incremental_alter_configs_resource>) {
+                auto op = static_cast<config_resource_operation>(
+                  c.config_operation);
+                if (op == config_resource_operation::set) {
+                    req.upsert.push_back(
+                      {ss::sstring(mapped_name), c.value.value()});
+                } else if (op == config_resource_operation::remove) {
+                    req.remove.push_back(ss::sstring(mapped_name));
+                } else {
+                    responses.push_back(
+                      make_error_alter_config_resource_response<R>(
+                        resource,
+                        error_code::invalid_config,
+                        fmt::format(
+                          "operation {} on broker properties isn't currently "
+                          "supported",
+                          op)));
+                    errored = true;
+                    continue;
+                }
+            } else {
+                if (c.value.has_value()) {
```

Comment:
This is the AlterConfigs path -- although Kafka API calls for this to reset all settings other than those in request, that's ambiguous for redpanda.  Caller probably doesn't want to reset all redpanda config options when they send a kafka-oriented request for e.g. topic defaults.

Reply:
I don't think it's worth it to support AlterConfigs considering it's a huge footgun and IncrementalAlterConfigs exists.

IncrementalAlterConfigs has existed since Kafka 2.3, so it's "relatively" modern considering a lot of people use older Kafka versions and stick on them. But, my take is we should raise the bar for Redpanda and only support the newer, safer API.